packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart

```diff
@@ -481,17 +416,27 @@ void _applyCameraUpdate(gmaps.GMap map, CameraUpdate update) {
 
 // original JS by: Byron Singh (https://stackoverflow.com/a/30541162)
 gmaps.LatLng _pixelToLatLng(gmaps.GMap map, int x, int y) {
-  final ne = map.bounds.northEast;
-  final sw = map.bounds.southWest;
+  final bounds = map.bounds;
   final projection = map.projection;
+  final zoom = map.zoom;
+
+  assert(
+      bounds != null, 'Map Bounds required to compute LatLng of screen x/y.');
+  assert(projection != null,
+      'Map Projection required to compute LatLng of screen x/y');
+  assert(zoom != null,
+      'Current map zoom level required to compute LatLng of screen x/y');
+
+  final ne = bounds!.northEast;
+  final sw = bounds.southWest;
 
-  final topRight = projection.fromLatLngToPoint(ne);
-  final bottomLeft = projection.fromLatLngToPoint(sw);
+  final topRight = projection!.fromLatLngToPoint!(ne)!;
+  final bottomLeft = projection.fromLatLngToPoint!(sw)!;
 
-  final scale = 1 << map.zoom; // 2 ^ zoom
+  final scale = 1 << (zoom! as int); // 2 ^ zoom
```

Comment:
What is the type of `zoom`? I'm curious why this isn't `.toInt()`

Reply:
This boi is `num?`. I did a quick test, and `.toInt()` is a much more robust cast:

```dart
void main() {
  num ber = 7.5;
  print(ber.toInt());
  print(ber as int);
}
```

Outputs:

```
7
Uncaught Error: TypeError: 7.5: type 'JSDouble' is not a subtype of type 'int'
```